### PR TITLE
CI: Fix yaml/yml inconsistency

### PR DESCRIPTION
**Pull request by jbrockmendel**
The ci/windows.yml and ci/windows-py27.yml files reference the environment files with ".yml".  This PR moves the ".yaml" files to the indicated location.

CI config is hard.

### CODE REVIEW

File: ci/incremental/setup_conda_environment.cmd [hunk 1]
```diff
@@ -13,7 +13,7 @@ conda list
 @rem Clean up any left-over from a previous build
 conda remove --all -q -y -n %CONDA_ENV%
 @rem Scipy, CFFI, jinja2 and IPython are optional dependencies, but exercised in the test suite
-conda env create -n %CONDA_ENV% --file=ci\azure-windows-%CONDA_PY%.yaml
+conda env create -n %CONDA_ENV% --file=ci\azure-windows-%CONDA_PY%.yml
```
Comment (jreback) [inline ci/incremental/setup_conda_environment.cmd]: why are you changing this? all of the other files are .yaml

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/incremental/setup_conda_environment.cmd (hunk 1 above)
Comment (jreback) [reply ci/incremental/setup_conda_environment.cmd]: would rather change the existing .yml -> .yaml

Comment (jbrockmendel) [reply ci/incremental/setup_conda_environment.cmd]: > all of the other files are .yaml

The top-level config files are .yml (e.g. .travis.yml)

> would rather change the existing .yml -> .yaml

Sure, will update.

File: ci/azure/macos.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@ jobs:
     maxParallel: 11
     matrix:
       py35_np_110:
-        ENV_FILE: ci/azure-macos-35.yml
```
Comment (jreback) [inline ci/azure/macos.yml]: is the name of this file crucial for azure? (e.g. the .yml vs .yaml)?

Comment (jbrockmendel) [reply ci/azure/macos.yml]: I honestly have no idea.  The fact that this name can be wrong without breaking the CI is troubling.

Comment (jreback) [reply ci/azure/macos.yml]: ok, can you rename these as well and see?

Comment (jbrockmendel) [reply ci/azure/macos.yml]: I must have miscommunicated.  The answer to the previous question is "apparently not".  In the status quo, the azure/windows-py27.yml and azure/windows.yml files have `ENV_FILE: ci/azure-windows-27.yml` that point to non-existent files, and azure doesn't appear to notice.

Comment (jreback) [reply ci/azure/macos.yml]: ahh interesting. so any idea how this was working? or is it just picking up the wrong env?

**Comment by jbrockmendel** [comment]
@TomAugspurger looks like you're the local expert on this.  Any ideas?

**Comment by TomAugspurger** [comment]
I guess azure is being clever and trying both? Regardless, standardizing on one seems good and  `.yaml` is more common.

**Pull request merged**